packages/lib-utils/src/components/details-page-header/DetailsPageHeader.tsx

```diff
@@ -0,0 +1,128 @@
+import {
+  Split,
+  SplitItem,
+  TextContent,
+  Text,
+  TextVariants,
+  Label,
+  DropdownPosition,
+} from '@patternfly/react-core';
+import * as _ from 'lodash';
+import React from 'react';
+import type { K8sResourceCommon } from '../../types/k8s';
+import type { BreadcrumbProp, ActionButtonProp, ActionMenuProps } from './utils';
+import { Breadcrumbs, ActionButtons, ActionMenu } from './utils';
+import '@patternfly/react-styles/css/utilities/Spacing/spacing.css';
+
+export type PageHeading = {
+  /** Optional title for page heading */
+  title?: string;
+  /** Optional label for page heading */
+  label?: {
+    name: string;
+    key?: string;
+    icon?: React.ReactNode;
+    href?: string;
+    color?: 'blue' | 'cyan' | 'green' | 'orange' | 'purple' | 'red' | 'grey';
+  };
+  /** Optional icon for page heading (appears to the left of the page heading's title) */
+  iconBeforeTitle?: React.ReactNode;
+  /** Optional icon for page heading (appears to the right of the page heading's title) */
+  iconAfterTitle?: React.ReactNode;
+};
+
+export type DetailsPageHeaderProps = {
+  breadcrumbs: BreadcrumbProp[];
+  actionButtons?: ActionButtonProp[];
+  pageHeading?: PageHeading;
+  /** Optional resource object (if no title for the page heading is provided, the title can be taken from the resource's name) */
+  obj?: K8sResourceCommon;
+  actionMenu?: ActionMenuProps;
+};
+
+export const DetailsPageHeader: React.SFC<DetailsPageHeaderProps> = ({
+  breadcrumbs,
+  actionButtons,
+  actionMenu,
+  pageHeading,
+  obj,
+}) => {
+  const heading = pageHeading?.title ?? obj?.metadata?.name;
+  return (
+    <>
+      <Split hasGutter className="pf-u-mb-sm">
```

Comment:
Also - not sure if a PatternFly layout could help with organizing some of these things...